examples/cross-contract-high-level/src/lib.rs

```diff
@@ -57,12 +35,12 @@ impl CrossContract {
         let pivot = arr.len() / 2;
         let arr0 = arr[..pivot].to_vec();
         let arr1 = arr[pivot..].to_vec();
-        let prepaid_gas = env::prepaid_gas();
         let account_id = env::current_account_id();
 
-        ext::merge_sort(arr0, account_id.clone(), 0, prepaid_gas / 4)
-            .and(ext::merge_sort(arr1, account_id.clone(), 0, prepaid_gas / 4))
-            .then(ext::merge(account_id, 0, prepaid_gas / 4))
+        Self::ext(account_id.clone())
```

Comment:
But they can call the self API to another contract. I thought you meant only in this case. The account_id can absolutely be different than the current.

I don't think it's worth adding another method for calling it on self, because cases like this you might want to avoid calling the host function more than once

Reply:
Another contract might have the same interface as the current contract, but I would personally propose `Self` always be calling the current contract.  It would make reading over the contract easier:

```rust
Self::merge_sort(arr0)
            .and(Self::merge_sort(arr1))
            .then(Self::merge())
```
And my point with `Self` handling the `account_id` is that it could lazily load and cache it.

Also You could still have `Self::ext` if you need the functionality, but for most uses of `Self` this would be a better DevX.

Reply:
This doesn't work. These methods already exist. Feel free to create a commit/diff if you have an idea for this working, and I'll happily consider it

Reply:
Ah silly me forgot it's the same `Self`, thought it was generated.  You would need to generate another module `current_contract`, which is a bit longer than `Self`.  I'll think about it more because this is a pain point.